clients/codelist/codelist_test.go

```diff
@@ -0,0 +1,1358 @@
+		}
+
+		codelistClient := &Client{
+			cli: clienter,
+			url: testHost,
+		}
+
+		Convey("when codelistClient.Healthcheck is called", func() {
+			actual, err := codelistClient.Healthcheck()
+
+			Convey("then the expected error is returned", func() {
+				So(actual, ShouldEqual, service)
+				So(err, ShouldResemble, expectedErr)
+			})
+
+			Convey("and client.Get should be called 1 time with the expected parameters", func() {
+				calls := clienter.GetCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].URL, ShouldEqual, testHost+uri)
+			})
+		})
+	})
+
+	Convey("given clienter.Get returns a non 200 response status", t, func() {
+		resp := httpmocks.NewResponseMock(nil, 401)
+
+		clienter := &rchttp.ClienterMock{
+			GetFunc: func(ctx context.Context, url string) (*http.Response, error) {
+				return resp, nil
+			},
+		}
+
+		codelistClient := &Client{
+			cli: clienter,
+			url: testHost,
+		}
+
+		Convey("when codelistClient.Healthcheck is called", func() {
+			actual, err := codelistClient.Healthcheck()
+
+			Convey("then the expected error is returned", func() {
+				So(actual, ShouldEqual, service)
+				So(err, ShouldResemble, &ErrInvalidCodelistAPIResponse{http.StatusOK, resp.StatusCode, "/healthcheck"})
+			})
+
+			Convey("and client.Get should be called 1 time with the expected parameters", func() {
+				calls := clienter.GetCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].URL, ShouldEqual, testHost+uri)
+			})
+		})
+	})
+
+	Convey("given clienter.Get returns a 200 response status", t, func() {
+		resp := httpmocks.NewResponseMock(nil, 200)
+
+		clienter := &rchttp.ClienterMock{
+			GetFunc: func(ctx context.Context, url string) (*http.Response, error) {
+				return resp, nil
+			},
+		}
+
+		codelistClient := &Client{
+			cli: clienter,
+			url: testHost,
+		}
+
+		Convey("when codelistClient.Healthcheck is called", func() {
+			actual, err := codelistClient.Healthcheck()
+
+			Convey("then no error is returned", func() {
+				So(actual, ShouldEqual, service)
+				So(err, ShouldBeNil)
+			})
+
+			Convey("and client.Get should be called 1 time with the expected parameters", func() {
+				calls := clienter.GetCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].URL, ShouldEqual, testHost+uri)
+			})
+		})
+	})
+}
+
+func TestClient_GetValues(t *testing.T) {
+
+	Convey("should return expect values for 200 status response", t, func() {
+		b, err := json.Marshal(testDimensionValues)
+		So(err, ShouldBeNil)
+
+		body := httpmocks.NewReadCloserMock(b, nil)
+		resp := httpmocks.NewResponseMock(body, 200)
+
+		clienter := getClienterMock(resp, nil)
+		codelistClient := &Client{cli: clienter, url: testHost}
+
+		actual, err := codelistClient.GetValues(nil, testServiceAuthToken, "999")
+
+		So(err, ShouldBeNil)
+		So(actual, ShouldResemble, testDimensionValues)
+
+		calls := clienter.DoCalls()
+		So(calls, ShouldHaveLength, 1)
+
+		req := calls[0].Req
+		So(req.URL.Path, ShouldEqual, "/code-lists/999/codes")
```

Comment:
So this:			

```
                                calls := clienter.DoCalls()
				So(calls, ShouldHaveLength, 1)

				req := calls[0].Req
				So(req.URL.Path, ShouldEqual, uri)
				So(req.URL.Host, ShouldEqual, host)
				So(req.Method, ShouldEqual, "GET")
				So(req.Body, ShouldBeNil)
				So(req.Header.Get(common.AuthHeaderKey), ShouldEqual, common.BearerPrefix+testServiceAuthToken)
```

Is repeated 25 times in the package, wondering if we can move this into its own helper function?